### ISSUE TITLE
Uncaught out of range exception

### ISSUE THREAD

**Issue by mtanana** [genielabs/zwave-lib-dotnet#28]
This issue crashes the zwave lib:

``` Trapper Unhandled Exception     Error.Exception System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) [0x00029] in <204f770036d441bb8dfd3daba3550e83>:0
  at System.ThrowHelper.ThrowArgumentOutOfRangeException () [0x00000] in <204f770036d441bb8dfd3daba3550e83>:0
  at System.Collections.Generic.List`1[T].RemoveAt (System.Int32 index) [0x00009] in <204f770036d441bb8dfd3daba3550e83>:0
  at ZWaveLib.CommandClasses.WakeUp.ResendOnWakeUp (ZWaveLib.ZWaveNode node, System.Byte[] msg) [0x0008c] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at ZWaveLib.ZWaveController.QueueMessage (ZWaveLib.ZWaveMessage message) [0x000b0] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at ZWaveLib.ZWaveNode.SendMessage (System.Byte[] message) [0x0000a] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at ZWaveLib.ZWaveNode.SendDataRequest (System.Byte[] request) [0x00039] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at ZWaveLib.CommandClasses.Version.Get (ZWaveLib.ZWaveNode node, ZWaveLib.CommandClass cmdClass) [0x00001] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at ZWaveLib.ZWaveController.GetNodeCcsVersion (ZWaveLib.ZWaveNode zn) [0x0006c] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at ZWaveLib.ZWaveController.Discovery () [0x00184] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at MIG.Interfaces.HomeAutomation.ZWave.Controller_ControllerStatusChanged (System.Object sender, ZWaveLib.ControllerStatusEventArgs args) [0x00072] in <1c438fb48a17499baf12c74b98118c9e>:0
  at ZWaveLib.ZWaveController.OnControllerStatusChanged (ZWaveLib.ControllerStatusEventArgs args) [0x00044] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at ZWaveLib.ZWaveController.Initialize () [0x00052] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at MIG.Interfaces.HomeAutomation.ZWave.Controller_ControllerStatusChanged (System.Object sender, ZWaveLib.ControllerStatusEventArgs args) [0x0002e] in <1c438fb48a17499baf12c74b98118c9e>:0
  at ZWaveLib.ZWaveController.OnControllerStatusChanged (ZWaveLib.ControllerStatusEventArgs args) [0x00044] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at ZWaveLib.ZWaveController.SerialPort_ConnectionStatusChanged (System.Object sender, SerialPortLib.ConnectionStatusChangedEventArgs args) [0x00033] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at SerialPortLib.SerialPortInput.OnConnectionStatusChanged (SerialPortLib.ConnectionStatusChangedEventArgs args) [0x0001f] in <fcc2756efd80414a84241ec0b47e73d4>:0
  at SerialPortLib.SerialPortInput.Open () [0x0014d] in <fcc2756efd80414a84241ec0b47e73d4>:0
  at SerialPortLib.SerialPortInput.Connect () [0x00029] in <fcc2756efd80414a84241ec0b47e73d4>:0
  at ZWaveLib.ZWaveController.<Connect>m__0 () [0x00001] in <0bdcd40ad78345dabdc1e0555226f3eb>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <204f770036d441bb8dfd3daba3550e83>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <204f770036d441bb8dfd3daba3550e83>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <204f770036d441bb8dfd3daba3550e83>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <204f770036d441bb8dfd3daba3550e83>:0
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <204f770036d441bb8dfd3daba3550e83>:0
```

**Comment by mtanana** [comment]
Just need a try catch here:   WakeUp.cs:134

**Comment by mtanana** [comment]
#29

**Comment by genemars** [comment]
merged #29
